romi-characterization-sysid/src/main/java/frc/robot/Robot.java

```diff
@@ -0,0 +1,261 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot;
+
+import java.util.ArrayList;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.livewindow.LiveWindow;
+import edu.wpi.first.wpilibj.romi.RomiGyro;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * The VM is configured to automatically run this class, and to call the
+ * functions corresponding to each mode, as described in the TimedRobot
+ * documentation. If you change the name of this class or the package after
+ * creating this project, you must also update the build.gradle file in the
+ * project.
+ */
+public class Robot extends TimedRobot {
+  private class QuasistaticVoltageSupplier implements Supplier<Double> {
+    private double rampRate = 0.0;
+    private double currentVoltage = 0.0;
+    private int counter = 0;
+
+    public QuasistaticVoltageSupplier(double rampRate) {
+      this.rampRate = rampRate;
+    }
+
+    @Override
+    public Double get() {
+      // Simple rate limiter. Without it Romi will keep running for at least
```

Comment:
Can you explain how this prevents the Romi from running after disabling? That should happen automatically. If I'm reading this correctly, This function creates a stepped ramp function (with step size 10*autonomous cycle).

Reply:
Right, that was my understanding too until I started observing this weird situation with my Romi running after disabling it. I wasn't able to figure out what is going on and how to fix it until I noticed that voltage in output generated by frc-characterization (which didn't have this issue) changes about every 10th cycles and when I tried this approach that fixed the issue.

PS: I didn't look into code that is running on Romi side but suspect that the issue may happen because of some sort of the queue that builds up if you change voltage too often.

Reply:
Reviewed, the queuing starts around .006 seconds. This example should be good for now. I'll merge